pulser/simulation/simulation.py

```diff
@@ -200,15 +210,15 @@ def _build_operator(self, op_id, *qubit_ids, global_op=False):
                    else self.op_matrix['I'] for j in range(self._size)]
         return qutip.tensor(op_list)
 
-    def _construct_hamiltonian(self):
-        def adapt(full_array):
+    def _construct_hamiltonian(self) -> None:
+        def adapt(full_array: np.ndarray) -> Any:
```

Comment:
Doesn't it return `np.ndarray` too?

Reply:
Yes, sorry I missed it.

Reply:
Try casting the `return` to `np.ndarray` then